langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -80,6 +89,16 @@ public isolated function cloneReadOnly(CloneableType  v) returns CloneableType &
 # - if a record type descriptor specifies default values, these will be used
 #   to supply any missing members
 #
+# ```
+# int[] numberArray = check [1, 2, 3, 4].cloneWithType();
+# numberArray ⇒ [1,2,3,4]
+#
+# json name = "Ballerina";
+# name.cloneWithType(string) ⇒ "Ballerina"
+#
+# name.cloneWithType(int) ⇒ error
```

Comment:
Better to have a different type instead of basic types since they don't really construct a value.